Code/GraphMol/Descriptors/BCUT.cpp

```diff
@@ -0,0 +1,97 @@
+//
+//  Copyright (C) 2020 Brian P. Kelley
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include "BCUT.h"
+#include <Eigen/Dense>
+#include <GraphMol/RDKitBase.h>
+#include "GraphMol/PartialCharges/GasteigerCharges.h"
+#include "GraphMol/PartialCharges/GasteigerParams.h"
+#include <RDGeneral/types.h>
+
+namespace RDKit {
+namespace Descriptors {
+  // diagonal elements are a property (atomic num, charge, etc)
+  // off diagonal are .1,.2,.3.15 for single,double,triple or aromatic
+  // all other elements are .001
+  std::pair<double,double> BCUT2D(const ROMol &m, const std::string &atom_double_prop) {
```

Comment:
How about having this take a vector of doubles with the values to use instead of the property name? That'd be a bit clearer and more efficient.
Can always keep an override that takes the property name and pulls it out into a vector of doubles